## Step-1:Data pre-processing

## Using SVMs(Support vector machines) algorithm

1. Importing the required libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

2. Importing the dataset using the pandas library

In [2]:
data=pd.read_csv("cell_samples.csv")
data

ID  Clump  UnifSize  UnifShape  MargAdh  SingEpiSize BareNuc  \
0    1000025      5         1          1        1            2       1   
1    1002945      5         4          4        5            7      10   
2    1015425      3         1          1        1            2       2   
3    1016277      6         8          8        1            3       4   
4    1017023      4         1          1        3            2       1   
..       ...    ...       ...        ...      ...          ...     ...   
694   776715      3         1          1        1            3       2   
695   841769      2         1          1        1            2       1   
696   888820      5        10         10        3            7       3   
697   897471      4         8          6        4            3       4   
698   897471      4         8          8        5            4       5   

     BlandChrom  NormNucl  Mit  Class  
0             3         1    1      2  
1             3         2    1      2  
2             3         1    1      2  
3             3         7    1      2  
4             3         1    1      2  
..          ...       ...  ...    ...  
694           1         1    1      2  
695           1         1    1      2  
696           8        10    2      4  
697          10         6    1      4  
698          10         4    1      4  

[699 rows x 11 columns]

In [3]:
data.shape

(699, 11)

In [4]:
data.dtypes

ID              int64
Clump           int64
UnifSize        int64
UnifShape       int64
MargAdh         int64
SingEpiSize     int64
BareNuc        object
BlandChrom      int64
NormNucl        int64
Mit             int64
Class           int64
dtype: object

# Here,BareNuc column contains the 'object' variable so we have to convert it in numerical variable

In [5]:
#Using to_numeric function of pandas library
data = data[pd.to_numeric(data['BareNuc'], errors='coerce').notnull()]
data['BareNuc'] = data['BareNuc'].astype('int')

#printing the results
data.dtypes

C:\Users\hp\AppData\Local\Temp\ipykernel_11744\1887118782.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['BareNuc'] = data['BareNuc'].astype('int')


ID             int64
Clump          int64
UnifSize       int64
UnifShape      int64
MargAdh        int64
SingEpiSize    int64
BareNuc        int32
BlandChrom     int64
NormNucl       int64
Mit            int64
Class          int64
dtype: object

3. Checking the missing values in the dataset

In [6]:
data.isnull().sum()

ID             0
Clump          0
UnifSize       0
UnifShape      0
MargAdh        0
SingEpiSize    0
BareNuc        0
BlandChrom     0
NormNucl       0
Mit            0
Class          0
dtype: int64

Here no any missing values found.

4. Extracting the independent and dependent variables

In [7]:
#for independent variable
x=data.iloc[:,[1,2,3,4,5,6,7,8,9]]
x.head(5)

Clump  UnifSize  UnifShape  MargAdh  SingEpiSize  BareNuc  BlandChrom  \
0      5         1          1        1            2        1           3   
1      5         4          4        5            7       10           3   
2      3         1          1        1            2        2           3   
3      6         8          8        1            3        4           3   
4      4         1          1        3            2        1           3   

   NormNucl  Mit  
0         1    1  
1         2    1  
2         1    1  
3         7    1  
4         1    1

In [8]:
#for dependent variable
y=data.iloc[:,10]
y.head(5)

0    2
1    2
2    2
3    2
4    2
Name: Class, dtype: int64

5. Splitting the dataset into Training and Training dataset

In [9]:
#importing the train_test_split class from the scikit-learn.model_selection library
from sklearn.model_selection import train_test_split

#splitting the dataset
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.25,random_state=0)

In [10]:
x_train

Clump  UnifSize  UnifShape  MargAdh  SingEpiSize  BareNuc  BlandChrom  \
571      7         6         10        5            3       10           9   
68       8         3          8        3            4        9           8   
586      8        10         10       10            6       10          10   
311      1         1          1        1            2        1           1   
369      1         1          3        1            1        1           2   
..     ...       ...        ...      ...          ...      ...         ...   
9        4         2          1        1            2        1           2   
373      5         1          1        2            2        1           2   
198      1         1          1        1            2        1           1   
645      3         1          1        1            2        1           2   
574     10         9          7        3            4        2           7   

     NormNucl  Mit  
571        10    2  
68          9    8  
586        10    1  
311         1    1  
369         1    1  
..        ...  ...  
9           1    1  
373         1    1  
198         1    1  
645         1    1  
574         7    1  

[512 rows x 9 columns]

In [11]:
x_test

Clump  UnifSize  UnifShape  MargAdh  SingEpiSize  BareNuc  BlandChrom  \
115      1         1          1        1            2        5           1   
392      3         1          1        1            2        1           2   
316      5         5          5        2            5       10           4   
519      4         7          8        3            4       10           9   
313      1         1          1        1            2        1           1   
..     ...       ...        ...      ...          ...      ...         ...   
458      5         1          2        1            2        1           1   
165      4         1          1        1            2        2           3   
331      5         1          1        1            2        1           3   
80       2         2          2        1            1        1           7   
94       2         1          1        1            2        1           3   

     NormNucl  Mit  
115         1    1  
392         1    1  
316         3    1  
519         1    1  
313         1    1  
..        ...  ...  
458         1    1  
165         2    1  
331         1    2  
80          1    1  
94          1    1  

[171 rows x 9 columns]

In [12]:
y_train

571    4
68     4
586    4
311    2
369    2
      ..
9      2
373    2
198    2
645    2
574    4
Name: Class, Length: 512, dtype: int64

In [13]:
y_test

115    2
392    2
316    4
519    4
313    2
      ..
458    2
165    2
331    2
80     2
94     2
Name: Class, Length: 171, dtype: int64

## Step-2: Fitting the SVM classifier to the training data:

In [14]:
#importing the kneighbors classifier form the scikit-learn.neighbors library
from sklearn import svm

#create an instance
classifier=svm.SVC(kernel='rbf')

#fitting the values in the classifier
classifier.fit(x_train,y_train)

SVC()

## Step-3:Predicting the test set result:

In [15]:
y_pred=classifier.predict(x_test)

In [16]:
y_pred

array([2, 2, 4, 4, 2, 2, 2, 4, 2, 2, 4, 2, 4, 2, 2, 4, 4, 4, 4, 2, 2, 2,
       4, 2, 4, 4, 2, 2, 2, 4, 2, 4, 4, 2, 2, 2, 4, 4, 2, 4, 2, 2, 2, 2,
       2, 2, 2, 4, 2, 2, 4, 2, 4, 2, 2, 2, 4, 4, 2, 4, 2, 2, 2, 2, 2, 2,
       2, 2, 4, 4, 2, 2, 2, 2, 2, 2, 4, 2, 2, 2, 4, 2, 4, 2, 2, 4, 2, 4,
       4, 2, 4, 2, 4, 4, 2, 4, 4, 4, 4, 2, 2, 2, 4, 4, 2, 2, 4, 4, 2, 2,
       4, 2, 2, 4, 2, 2, 2, 4, 2, 2, 2, 4, 2, 2, 4, 4, 2, 4, 2, 4, 2, 2,
       4, 2, 2, 4, 2, 4, 2, 2, 2, 4, 2, 2, 2, 4, 4, 2, 4, 2, 4, 2, 2, 2,
       2, 2, 4, 4, 2, 4, 4, 4, 4, 2, 4, 2, 2, 2, 2, 2, 2], dtype=int64)

## Step-4:Creating the confusion matrix

In [17]:
#importing the confusion matrix class from the scikit-learn.metrics library
from sklearn.metrics import confusion_matrix

#creating an instance
cm=confusion_matrix(y_test,y_pred)

In [18]:
cm

array([[102,   5],
       [  3,  61]], dtype=int64)

## Step-5:Printing the classification report

In [19]:
#importing the classification_report class from scikit-learn library.
from sklearn.metrics import classification_report

#printing the results
print(classification_report(y_test,y_pred))

              precision    recall  f1-score   support

           2       0.97      0.95      0.96       107
           4       0.92      0.95      0.94        64

    accuracy                           0.95       171
   macro avg       0.95      0.95      0.95       171
weighted avg       0.95      0.95      0.95       171



## Step-6:Accuracy score

In [20]:
from sklearn.metrics import accuracy_score

#creating an instance
accuracy=accuracy_score(y_test,y_pred)

#printing the accuracy score
print("Accuracy score:",accuracy*100,"%")

Accuracy score: 95.32163742690058 %


## Hence the accuracy score of SVM model is 95.32%